src/frontend/src/pages/new-challenge/new-challenge.tsx

```diff
@@ -1,7 +1,121 @@
+                    1 - Select your word
+                </Typography>
+                <div style={{ padding: 50, alignSelf: "center" }}>
+                    <Card>
+                        <CardContent>
+                            <Typography gutterBottom>Random Word</Typography>
+                        </CardContent>
+                        <CardActions>
+                            {/* Create an 'onClick' method that calls the API */}
+                            <Button size="small" color="primary">
+                                Change
+                            </Button>
+                        </CardActions>
+                    </Card>
+                </div>
+                {/*  spacing */}
+                <div style={{ padding: 50 }}>
+                    <Typography gutterBottom variant="h4">
+                        2 - Select your definitions
+                    </Typography>
+                </div>
+                <Grid
+                    container
+                    direction="row"
+                    justify="space-evenly"
+                    alignItems="stretch"
+                    spacing={3}
+                >
+                    <Grid item xs={4}>
+                        <Typography>Actual Definition</Typography>
+                    </Grid>
+                    <Grid item xs={4}>
+                        <Card>
+                            <CardContent>
+                                <Typography gutterBottom>
+                                    Optional Definition1
+                                </Typography>
+                            </CardContent>
+                            <CardActions>
+                                {/* Create an 'onClick' method that calls the API */}
+                                <Button size="small" color="primary">
+                                    Change
+                                </Button>
+                            </CardActions>
+                        </Card>
+                    </Grid>
+                    <Grid item xs={4}>
+                        <Card>
+                            <CardContent>
+                                <Typography gutterBottom>
+                                    Optional Definition2
+                                </Typography>
+                            </CardContent>
+                            <CardActions>
+                                {/* Create an 'onClick' method that calls the API */}
+                                <Button size="small" color="primary">
+                                    Change
+                                </Button>
+                            </CardActions>
+                        </Card>
+                    </Grid>
+                </Grid>
+                {/*  spacing */}
+                <div style={{ padding: 50 }}></div>
+                <Typography gutterBottom variant="h4">
+                    3 - Input your sentence
+                </Typography>
```

Comment:
```suggestion
                <Box p={3}>
                    <Typography gutterBottom variant="h4">
                        3 - Input your sentence
                    </Typography>
                </Box>
```
Please wrap these components with a box with spacing instead of using what is effectively a <br /> element. You can use px={3} or whatever to set padding to the top and bottom only if that is the desired effect.

Reply:
I appreciate if you could clarify why the recommended way would be more efficient.

Reply:
Sure. You shouldn't really be adding block elements at the same level to introduce margin or padding. Instead, you should either adding the margin or padding to that component, or wrap it in a "higher order component" that adds it. This way, the margin and padding "belongs" to the component rather than floating somewhere before or after it in the component tree.